code/modules/mob/living/simple_animal/abnormality/_abnormality.dm

```diff
@@ -261,18 +263,17 @@
 
 // Called by datum_reference when work is done
 /mob/living/simple_animal/hostile/abnormality/proc/work_complete(mob/living/carbon/human/user, work_type, pe, work_time, canceled)
```

Comment:
Since you moved blue star's parent call to the end, might want to add some checks here for user not being deleted or dead.